.travis.yml

```diff
@@ -1,49 +1,28 @@
 language: c
 
-env:
-  matrix:
-    - BD=esp8266:esp8266:d1_mini:xtal=80,eesz=4M3M
-    - BD=esp32:esp32:esp32
-  
-before_install:
-  - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_1.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :1 -ac -screen 0 1280x1024x16"
-  - sleep 3
-  - export DISPLAY=:1.0
-  - wget http://downloads.arduino.cc/arduino-1.8.8-linux64.tar.xz
-  - tar xf arduino-*-linux64.tar.xz
-  - export PATH=$(readlink -f arduino-*/):$PATH
-  
 install:
-  - arduino --pref "boardsmanager.additional.urls=http://arduino.esp8266.com/stable/package_esp8266com_index.json,https://dl.espressif.com/dl/package_esp32_index.json" --save-prefs
-  - if [[ "$BD" =~ "esp8266:esp8266:" ]]; then arduino --install-boards esp8266:esp8266:2.5.0 ; fi
-  - if [[ "$BD" =~ "esp32:esp32:" ]]; then arduino --install-boards esp32:esp32 ; fi
-  - arduino --pref "compiler.warning_level=all" --save-prefs
-  - # For async; unfortunately these libraries are still not the default in arduino/esp8266 yet
-  - mkdir -p $HOME/Arduino/libraries/
-  - cd $HOME/Arduino/libraries/
-  - git clone https://github.com/greiman/SdFat
-  - git clone -o 95dedf7 https://github.com/me-no-dev/ESPAsyncWebServer
-  - git clone -o 7e9ed22 https://github.com/me-no-dev/ESPAsyncTCP # for esp8266
-  - git clone https://github.com/me-no-dev/AsyncTCP # for esp32
-  - git clone https://github.com/ayushsharma82/AsyncElegantOTA
-  - git clone -o 1c02154 https://github.com/alanswx/ESPAsyncWiFiManager
-  - ( git clone -b 6.x https://github.com/bblanchon/ArduinoJson ; cd ArduinoJson ; git reset --hard 3df4efd )
-  - git clone -o 9619fef https://github.com/Makuna/NeoPixelBus # https://github.com/Makuna/NeoPixelBus/issues/275
-  - if [[ "$BD" =~ "esp32:esp32:" ]]; then git clone https://github.com/bbx10/Hash_tng ; fi
-  - cd -
+  - sudo apt-get update && sudo apt-get install -y --no-install-recommends python3-minimal python3-pip python3-setuptools
+  - pip3 install platformio==4.2.1
```

Comment:
I just pinned it because you prefer that it seems. I don't think there'll be issues with any PlatformIO version `>= 4`, though. Pinning the dependencies makes sense indeed.